File: backend/src/main/java/com/nanmuli/blog/domain/project/Project.java
package com.nanmuli.blog.domain.project;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.nanmuli.blog.infrastructure.config.db.JsonbTypeHandler;
import com.nanmuli.blog.shared.domain.BaseAggregateRoot;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@EqualsAndHashCode(callSuper = false)
@TableName("project_showcase")
public class Project extends BaseAggregateRoot<Long> {
    private static final long serialVersionUID = 1L;

    private String name;
    private String slug;
    private String description;
    private String cover;

    @TableField(typeHandler = JsonbTypeHandler.class)
    private List<String> screenshots;

    @TableField(typeHandler = JsonbTypeHandler.class)
    private List<String> techStack;

    private String githubUrl;
    private String demoUrl;
    private String docUrl;
    private Integer sort;
    private Integer status;
    private LocalDate startDate;
    private LocalDate endDate;
}

File: frontend/src/types/project.ts
// 注意：ID 使用 string 类型，因为 JavaScript number 无法精确表示 64 位整数
export interface Project {
  id: string
  name: string
  slug: string
  description: string
  cover: string
  screenshots: string[]
  techStack: string[]
  githubUrl: string
  demoUrl: string
  docUrl: string
  sort: number
  status: number
  startDate: string
  endDate: string
  createTime: string
  updateTime: string
}
